mono/metadata/w32event-unix.c

```diff
@@ -376,15 +376,15 @@ ves_icall_System_Threading_Events_ResetEvent_internal (gpointer handle)
 
 	event_handle->set_count = 0;
 
-	mono_w32handle_unlock_handle (handle);
+	mono_w32handle_unlock_handle (handle, MONO_W32HANDLE_EVENT);
 
 	return TRUE;
 }
 
 void
 ves_icall_System_Threading_Events_CloseEvent_internal (gpointer handle)
 {
-	mono_w32handle_close (handle);
+	mono_w32handle_close (handle, MONO_W32HANDLE_ANY);
```

Comment:
I got `MONO_W32HANDLE_PROCESS` here before (please excuse the multiprocess output mess):

```mono_w32handle_unref: w32handle has unexpected type 0x9, allowed types must match mask 0x6
mono_w32handle_unref: w32handle has unexpected type 0x9, allowed types must match mask 0x6
mono_w32handle_unref: w32handle has unexpected type 0x9, allowed types must match mask 0x6
Stacktrace:

mono_w32handle_unref: w32handle has unexpected type 0x9, allowed types must match mask 0x6
  at <unknown> <0xffffffff>
mono_w32handle_unref: w32handle has unexpected type 0x9, allowed types must match mask 0x6
  at (wrapper managed-to-native) System.Threading.NativeEventCalls.CloseEvent_internal (intptr) [0x00000] in <d16181e4d6d14ea0b9f437c2e12ca42e>:0
  at Microsoft.Win32.SafeHandles.SafeWaitHandle.ReleaseHandle () [0x00000] in <d16181e4d6d14ea0b9f437c2e12ca42e>:0
  at System.Runtime.InteropServices.SafeHandle.DangerousReleaseInternal (bool) [0x000a3] in <d16181e4d6d14ea0b9f437c2e12ca42e>:0
  at System.Runtime.InteropServices.SafeHandle.InternalDispose () [0x0000e] in <d16181e4d6d14ea0b9f437c2e12ca42e>:0
  at System.Runtime.InteropServices.SafeHandle.Dispose (bool) [0x00003] in <d16181e4d6d14ea0b9f437c2e12ca42e>:0
  at System.Runtime.InteropServices.SafeHandle.Close () [0x00000] in <d16181e4d6d14ea0b9f437c2e12ca42e>:0
  at System.Threading.WaitHandle.Dispose (bool) [0x0000a] in <d16181e4d6d14ea0b9f437c2e12ca42e>:0
  at System.Threading.WaitHandle.Close () [0x00000] in <d16181e4d6d14ea0b9f437c2e12ca42e>:0
  at System.Diagnostics.Process.WaitForExit (int) [0x00075] in <9cea9f98604046508e9499a58e7c0ac8>:0
  at (wrapper remoting-invoke-with-check) System.Diagnostics.Process.WaitForExit (int) [0x00032] in <9cea9f98604046508e9499a58e7c0ac8>:0```